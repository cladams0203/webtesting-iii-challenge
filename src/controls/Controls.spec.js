// Test away!
import React from 'react'
import Controls from './Controls';
import { render } from 'react-testing-library';
import '@testing-library/jest-dom/extend-expect';

test('Controls renders without crashing', () => {
    render(<Controls/>)
})
test('Controls Snapshot', () => {
    const wrapper = render(<Controls/>)
    expect(wrapper.asFragment()).toMatchSnapshot()
})
test('Buttons display in document', () => {
    const wrapper = render(<Controls/>)
    const button1 = wrapper.getByTestId('button1')
    const button2 = wrapper.getByTestId('button2')
    expect(button1).toBeInTheDocument()
    expect(button2).toBeInTheDocument()
})
test('Button text reflects state', () => {
    const wrapper = render(<Controls/>)
    const button1 = wrapper.getByTestId('button1')
    const lockbutton = button1.getAttribute('disabled')
    lockbutton === false ? 
    expect(button1.textContent).toMatch(/unlock gate/i) :
    expect(button1.textContent).toMatch(/lock gate/i)
    const button2 = wrapper.getByTestId('button2')
    const lockbutton2 = button2.getAttribute('disabled')
    lockbutton2 === false ? 
    expect(button2.textContent).toMatch(/open gate/i) :
    expect(button2.textContent).toMatch(/close gate/i)

})


