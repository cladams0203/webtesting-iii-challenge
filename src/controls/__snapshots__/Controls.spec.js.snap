// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Snapshot 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="button1"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="button2"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
